include/seqan3/range/detail/misc.hpp

```diff
@@ -40,4 +40,34 @@ template <std::ranges::forward_range rng_t>
 constexpr void consume(rng_t &&)
 {}
 
+/*!\brief Mimics std::ranges::next(it, s, bound) but additionally returns how often `it` was incremented.
+ * \ingroup range
+ * \tparam iterator_t Type of the iterator; must model std::ranges::forward_iterator.
+ * \tparam size_type Type of the size; must model arithmetic.
+ * \tparam sentinal_t Type of the range; must model std::ranges::sentinel_for<sentinal_t, iterator_t>.
+ * \param[in] it The iterator to increment.
+ * \param[in] s The number of steps to increment.
+ * \param[in] sen The sentinel used as a bound for incrementing `it`.
+ * \returns a std::pair over the incremented iterator and how often `it` was incremented.
+ */
+template <std::ranges::forward_iterator iterator_t, arithmetic size_type, typename sentinal_t>
```

Comment:
```suggestion
template <std::ranges::forward_iterator iterator_t, std::integral size_type, typename sentinal_t>
```

You cannot increment by `0.4` I think :wink: